File: Aplicativo/OndeAlmocar/src/br/com/ondealmocar/Filtro.java
package br.com.ondealmocar;

import java.math.BigDecimal;

import org.json.JSONException;
import org.json.JSONObject;

public class Filtro implements ConversorJson {
	
  private List<String> tipos;
  private Boolean salaoProprio;
  private String quantPessoas;
  private List<String> favoritos;
  private Boolean buscaExata;
  private Double latitude;
  private Double longitude;
  
  
  public String json() {
		JSONObject jsonObject = new JSONObject();
		try {
		  jsonObject.put("tipos", this.tipos);
		  jsonObject.put("quant_pessoas", this.quantPessoas);
		  jsonObject.put("favoritos", this.favoritos);
		  jsonObject.put("busca_exata", this.buscaExata);
		  jsonObject.put("latitude", this.latitude);
		  jsonObject.put("longitude", this.longitude);
		} catch (JSONException e) {
			e.printStackTrace();
		}
		return jsonObject.toString();
	}

	public void entidade(String json) {
		try {
			JSONObject jsonObject = new JSONObject(json);
			this.precoMedio = new BigDecimal(jsonObject.optDouble("preco_medio"));
			//TODO TERMINAR....
		} catch (JSONException e) {
			e.printStackTrace();
		}

	}
  }

File: BackEnd/ondealmocar/controladores/filtro.py
# -*- coding: utf-8 -*-

class Filtro():

    global preco_medio
    global tipos
    global salao_proprio
    global quant_pessoas
    global nivel_silencio
    global nivel_conforto
    global favoritos
    global busca_exata
    global latitude
    global longitude
    
    def __init__(self):
        self.preco_medio = None
        self.tipos = []
        self.salao_proprio = None
        self.quant_pessoas = None
        self.nivel_silencio = None
        self.nivel_conforto = None
        self.favoritos = None
        self.busca_exata = None
        self.latitude = 0
        self.longitude = 0
    
    def montarFiltro(self, json_data):
        self.preco_medio = json_data['preco_medio']
        self.tipos = json_data['tipos']
        self.salao_proprio = json_data['salao_proprio']
        self.quant_pessoas = json_data['quant_pessoas']
        self.nivel_silencio = json_data['nivel_silencio']
        self.nivel_conforto = json_data['nivel_conforto']
        self.favoritos = json_data['favoritos']
        self.busca_exata = json_data['busca_exata']
        self.latitude = json_data['latitude']
        self.longitude = json_data['longitude']
        return self
